### ISSUE TITLE
Updating multiple beanstalk variables in the same apply causes a crash

### ISSUE THREAD

**Issue by fluzz** [hashicorp/terraform#9259]
### Terraform Version
``` 
terraform -v
Terraform v0.7.4
```

### Affected Resource(s)
aws_elastic_beanstalk_environment

### Terraform Configuration Files

original
```
resource "aws_elastic_beanstalk_application" "tftest" {
  name = "tf-test-name"
  description = "tf-test-desc"
}

resource "aws_elastic_beanstalk_environment" "test" {
  name = "test-name"
  application = "${aws_elastic_beanstalk_application.tftest.name}" 
  solution_stack_name = "64bit Amazon Linux 2016.03 v2.1.6 running Python 3.4"
  setting {
    namespace = "aws:elasticbeanstalk:application:environment"
    name = "ENV_BAR"
    value = "BAR"
  }
  setting {
    namespace = "aws:elasticbeanstalk:application:environment"
    name = "ENV_FOO"
    value = "BAR"
  }
}

```

updated
```
resource "aws_elastic_beanstalk_application" "tftest" {
  name = "tf-test-name"
  description = "tf-test-desc"
}

resource "aws_elastic_beanstalk_environment" "test" {
  name = "test-name"
  application = "${aws_elastic_beanstalk_application.tftest.name}" 
  solution_stack_name = "64bit Amazon Linux 2016.03 v2.1.6 running Python 3.4"
  setting {
    namespace = "aws:elasticbeanstalk:application:environment"
    name = "ENV_BAR"
    value = "BAZ"
  }
  setting {
    namespace = "aws:elasticbeanstalk:application:environment"
    name = "ENV_FOO"
    value = "BAZ"
  }
}
```

### Panic Output
https://gist.github.com/fluzz/338bdbd233d8fa2f6f9a1973bd1d94f5

### Expected Behavior
An update to the two beanstalk environment variables

### Actual Behavior
See crash.log

corrupts remote state file causing manual reverted to pre-existing version

### Steps to Reproduce

1. change multiple variable in the beanstalk configuration
1. `terraform apply`

### Important Factoids
updating single variables works fine

**Comment by stack72** [comment]
Relevant panic output:

```
2016/10/06 17:13:14 [DEBUG] plugin: terraform: panic: runtime error: slice bounds out of range
2016/10/06 17:13:14 [DEBUG] plugin: terraform: 
2016/10/06 17:13:14 [DEBUG] plugin: terraform: goroutine 235 [running]:
2016/10/06 17:13:14 [DEBUG] plugin: terraform: panic(0x249f2c0, 0xc42000e0d0)
2016/10/06 17:13:14 [DEBUG] plugin: terraform: 	/usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
2016/10/06 17:13:14 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/builtin/providers/aws.resourceAwsElasticBeanstalkEnvironmentUpdate(0xc420774d20, 0x21fce60, 0xc42049c600, 0x25285c0, 0x15d81)
2016/10/06 17:13:14 [DEBUG] plugin: terraform: 	/private/tmp/terraform-20160920-35578-1ycpsj5/terraform-0.7.4/src/github.com/hashicorp/terraform/builtin/providers/aws/resource_aws_elastic_beanstalk_environment.go:358 +0x72d
2016/10/06 17:13:14 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc420570b40, 0xc4204ee000, 0xc4205acb20, 0x21fce60, 0xc42049c600, 0x1, 0xc4208d3cb0, 0x0)
2016/10/06 17:13:14 [DEBUG] plugin: terraform: 	/private/tmp/terraform-20160920-35578-1ycpsj5/terraform-0.7.4/src/github.com/hashicorp/terraform/helper/schema/resource.go:168 +0x1ff
2016/10/06 17:13:14 [DEBUG] plugin: terraform: github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc42057e2d0, 0xc420766340, 0xc4204ee000, 0xc4205acb20, 0x0, 0x18, 0x18)
2016/10/06 17:13:14 [DEBUG] plugin: terraform: 	/private/tmp/terraform-20160920-35578-1ycpsj5/terraform-0.7.4/src/github.com/hashicorp/terraform/helper/schema/provider.go:178 +0x9b
2016
```

**Comment by fluzz** [comment]
Still occurs when running 0.7.5

**Comment by ninjaMog** [comment]
Also encountering this issue.  It's a major pain point for us as we now have to carefully upgrade settings one by one, or work around it by making changes in the console.

The remote state getting corrupted is terrifying, undesirable changes appeared on a replan afterwards.  I'm not sure what affect an apply would have had.

Luckily we had versions enabled on our remote bucket so we could rollback the state, but this is very time consuming.

**Comment by cartolari** [comment]
I can confirm this too. Although I keep the state files versioned, I find easier, if only the aws_elastic_beanstalk_environment got affected, to do a terraform import with the environment id, which generally works and Terraform can continue from there.

**Comment by fluzz** [comment]
Still fails with 0.7.6 and while we can simply reimport the beanstalk in this case our production setups there a multiple resources that need reimporting

**Comment by pmacdougall** [comment]
I'm seeing the same behavior even with no changes in the configuration.  Using version 0.7.6 and a new beanstalk environment.  Just running apply again right after the initial create with no changed causes this error for me.

**Comment by fromonesrc** [comment]
This is _extremely_ painful for us as heavy users of Elastic Beanstalk.

**Comment by myoung34** [comment]
I tested this against #8983 and it resolved it for me

**Comment by fromonesrc** [comment]
Awesome!

**Comment by tecnobrat** [comment]
Looks like #8983 has been merged, so this should be fixed?

**Comment by stack72** [comment]
Closed via #8983
